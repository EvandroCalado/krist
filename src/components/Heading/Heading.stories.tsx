import { Meta, StoryFn } from '@storybook/react';

import { Heading, HeadingProps } from './Heading';

export default {
  title: 'Components/Heading',
  component: Heading,
  args: {
    children: 'Heading',
  },
} as Meta;

export const Default: StoryFn<HeadingProps> = (args) => <Heading {...args} />;
