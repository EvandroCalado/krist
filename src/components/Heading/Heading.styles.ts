import styled, { css } from 'styled-components';

import { HeadingProps } from '.';

export const Title = styled.h1<HeadingProps>`
  ${({
    theme,
    size = 'md',
    transform = 'lowercase',
    align = 'left',
    fontWeight = 'normal',
  }) => css`
    font-size: ${theme.font.sizes[size]};
    text-transform: ${transform};
    text-align: ${align};
    font-weight: ${fontWeight};
  `}
`;
