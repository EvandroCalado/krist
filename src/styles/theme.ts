export default {
  colors: {
    primary: '#1d4ed8',
    secondary: '#dc2626',
    warning: '#e11d48',
    success: '#16a34a',
    info: '#3b82f6',
    white: '#ffffff',
    black: '#000000',
    background: '#f5f5f5',
    text: '#262626',
  },
  font: {
    family: {
      primary: "'Inter', sans-serif;",
    },
    sizes: {
      xs: '.8rem',
      sm: '1.4rem',
      md: '2.4rem',
      lg: '3.2rem',
      xl: '4rem',
      '2xl': '4.8rem',
      '3xl': '5.6rem',
      '4xl': '6.4rem',
      '5xl': '7.2rem',
      '6xl': '8.0rem',
      '7xl': '9.6rem',
    },
  },
  media: {
    lteOrEqSmallest: '(max-width: 320px)',
    lteOrEqSmall: '(max-width: 576px)',
    lteOrEqMedium: '(max-width: 768px)',
    lteOrEqLarge: '(max-width: 992px)',
    gteOrEqSmallest: '(min-width: 320px)',
    gteOrEqSmall: '(min-width: 576px)',
    gteOrEqMedium: '(min-width: 768px)',
    gteOrEqLarge: '(min-width: 992px)',
  },
  spacings: {
    xs: '0.5rem',
    sm: '1rem',
    md: '2rem',
    lg: '3rem',
    xl: '4rem',
    '2xl': '5rem',
    '3xl': '6rem',
    '4xl': '7rem',
    '5xl': '8rem',
    '6xl': '9rem',
    '7xl': '10rem',
  },
  frameSizes: {
    xs: '8rem',
    sm: '12rem',
    md: '24rem',
    lg: '32rem',
    xl: '40rem',
    '2xl': '56rem',
    '3xl': '64rem',
    '4xl': '80rem',
    '5xl': '104rem',
    '6xl': '120rem',
    '7xl': '134rem',
  },
  transitions: {
    fastest: 'all 100ms ease-in-out',
    faster: 'all 200ms ease-in-out',
    fast: 'all 300ms ease-in-out',
    normal: 'all 500ms ease-in-out',
    slow: 'all 600ms ease-in-out',
    slower: 'all 800ms ease-in-out',
    slowest: 'all 1s ease-in-out',
  },
  layers: {
    normal: 0,
    layer1: 1,
    layer2: 2,
    layer3: 3,
    layer4: 4,
    layer5: 5,
    layer6: 6,
    layer7: 7,
    layer8: 8,
    layer9: 9,
    aboveAll: 10,
  },
} as const;
