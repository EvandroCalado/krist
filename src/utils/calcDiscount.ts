export const calcDiscount = (value: number, discount: number) => {
  const valueWithDiscount = (value * (100 - discount)) / 100;

  if (discount === 0) return value;

  return valueWithDiscount.toFixed(2);
};
