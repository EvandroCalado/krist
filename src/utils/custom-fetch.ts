import axios from 'axios';

export const customFetch = axios.create({
  baseURL: import.meta.env.VITE_STRAPI_URL,
});
