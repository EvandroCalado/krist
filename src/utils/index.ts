export * from './calcDiscount';
export * from './custom-fetch';
export * from './formatDate';
export * from './linksProfileMenu';
export * from './test-utils';
