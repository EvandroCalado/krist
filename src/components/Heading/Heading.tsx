import { FC } from 'react';

import * as S from './Heading.styles';

export type HeadingProps = {
  children: React.ReactNode;
  as?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
  size?: 'md' | 'lg' | 'xl' | '2xl' | '3xl' | '4xl' | '5xl' | '6xl' | '7xl';
  transform?: 'uppercase' | 'lowercase' | 'capitalize';
  align?: 'left' | 'center' | 'right';
};

export const Heading: FC<HeadingProps> = ({
  children,
  as = 'h1',
  size = 'md',
  transform = 'lowercase',
  align = 'left',
}) => {
  return (
    <S.Container as={as} size={size} transform={transform} align={align}>
      {children}
    </S.Container>
  );
};
