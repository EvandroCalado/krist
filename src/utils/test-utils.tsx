import { render, RenderOptions } from '@testing-library/react';
import { ReactElement } from 'react';
import { ThemeProvider } from 'styled-components';

import theme from 'styles/theme';

type CustomRenderProps = Omit<RenderOptions, 'queries'>;

const customRender = (
  ui: ReactElement,
  { ...renderOptions }: CustomRenderProps = {},
) => render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, renderOptions);

export * from '@testing-library/react';
export { customRender as render };
