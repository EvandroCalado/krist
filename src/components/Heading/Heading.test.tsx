import { screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';

import { customRender } from 'utils/test-utils';

import theme from 'styles/theme';

import { Heading } from './Heading';

describe('Heading', () => {
  it('should render default', () => {
    customRender(<Heading>Heading</Heading>);

    const heading = screen.getByRole('heading');

    expect(heading).toBeInTheDocument();
    expect(heading).toHaveTextContent('Heading');
    expect(heading).toHaveStyle({
      'font-size': theme.font.sizes.md,
      'text-transform': 'lowercase',
      'text-align': 'left',
    });
  });

  it('should render transform', () => {
    const { rerender } = customRender(
      <Heading transform="uppercase">Heading</Heading>,
    );

    const heading = screen.getByRole('heading');

    expect(heading).toHaveStyle({
      'text-transform': 'uppercase',
    });

    rerender(
      <ThemeProvider theme={theme}>
        <MemoryRouter>
          <Heading transform="lowercase">Heading</Heading>
        </MemoryRouter>
      </ThemeProvider>,
    );

    expect(heading).toHaveStyle({
      'text-transform': 'lowercase',
    });

    rerender(
      <ThemeProvider theme={theme}>
        <MemoryRouter>
          <Heading transform="capitalize">Heading</Heading>
        </MemoryRouter>
      </ThemeProvider>,
    );

    expect(heading).toHaveStyle({
      'text-transform': 'capitalize',
    });
  });

  it('should render sizes', () => {
    const { rerender } = customRender(<Heading size="lg">Heading</Heading>);

    const heading = screen.getByRole('heading');

    expect(heading).toHaveStyle({
      'font-size': theme.font.sizes.lg,
    });

    rerender(
      <ThemeProvider theme={theme}>
        <MemoryRouter>
          <Heading size="xl">Heading</Heading>
        </MemoryRouter>
      </ThemeProvider>,
    );

    expect(heading).toHaveStyle({
      'font-size': theme.font.sizes.xl,
    });

    rerender(
      <ThemeProvider theme={theme}>
        <MemoryRouter>
          <Heading size="2xl">Heading</Heading>
        </MemoryRouter>
      </ThemeProvider>,
    );

    expect(heading).toHaveStyle({
      'font-size': theme.font.sizes['2xl'],
    });

    rerender(
      <ThemeProvider theme={theme}>
        <MemoryRouter>
          <Heading size="3xl">Heading</Heading>
        </MemoryRouter>
      </ThemeProvider>,
    );

    expect(heading).toHaveStyle({
      'font-size': theme.font.sizes['3xl'],
    });

    rerender(
      <ThemeProvider theme={theme}>
        <MemoryRouter>
          <Heading size="4xl">Heading</Heading>
        </MemoryRouter>
      </ThemeProvider>,
    );

    expect(heading).toHaveStyle({
      'font-size': theme.font.sizes['4xl'],
    });

    rerender(
      <ThemeProvider theme={theme}>
        <MemoryRouter>
          <Heading size="5xl">Heading</Heading>
        </MemoryRouter>
      </ThemeProvider>,
    );

    expect(heading).toHaveStyle({
      'font-size': theme.font.sizes['5xl'],
    });

    rerender(
      <ThemeProvider theme={theme}>
        <MemoryRouter>
          <Heading size="6xl">Heading</Heading>
        </MemoryRouter>
      </ThemeProvider>,
    );

    expect(heading).toHaveStyle({
      'font-size': theme.font.sizes['6xl'],
    });

    rerender(
      <ThemeProvider theme={theme}>
        <MemoryRouter>
          <Heading size="7xl">Heading</Heading>
        </MemoryRouter>
      </ThemeProvider>,
    );

    expect(heading).toHaveStyle({
      'font-size': theme.font.sizes['7xl'],
    });
  });

  it('should render element correct', () => {
    const { container } = customRender(<Heading as="h3">Heading</Heading>);

    const h3 = container.querySelector('h3')!;

    expect(h3.tagName.toLowerCase()).toBe('h3');
  });

  it('should render with align correct', () => {
    const { rerender } = customRender(
      <Heading align="center">Heading</Heading>,
    );

    const heading = screen.getByRole('heading');

    expect(heading).toHaveStyle({
      'text-align': 'center',
    });

    rerender(
      <ThemeProvider theme={theme}>
        <MemoryRouter>
          <Heading align="right">Heading</Heading>
        </MemoryRouter>
      </ThemeProvider>,
    );

    expect(heading).toHaveStyle({
      'text-align': 'right',
    });
  });

  it('should render to match snapshot', () => {
    const { container } = customRender(<Heading as="h3">Heading</Heading>);

    expect(container).toMatchSnapshot();
  });
});
