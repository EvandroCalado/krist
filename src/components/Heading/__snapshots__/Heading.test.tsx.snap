// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should render to match snapshot 1`] = `
.c0 {
  font-size: 1.6rem;
  text-transform: lowercase;
  text-align: left;
  font-weight: normal;
}

<div>
  <h3
    class="c0"
  >
    Heading
  </h3>
</div>
`;
